jupyter_sparkmonitor/sparkmonitor.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+# Copyright 2018 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This init script installs Apache Zeppelin on the master node of a Cloud
+# Dataproc cluster. Zeppelin is also configured based on the size of your
+# cluster and the versions of Spark/Hadoop which are installed.
+
+set -euxo pipefail
+
+readonly ROLE="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
+
+readonly JUPYTER_INIT_SCRIPT='/usr/lib/systemd/system/jupyter.service'
+readonly CONDA_DIRECTORY='/opt/conda/default'
+readonly PYTHON_PATH="${CONDA_DIRECTORY}/bin/python"
+readonly EXISTING_PYSPARK_KERNEL='pyspark'
+
+function retry_apt_command() {
+  cmd="$1"
+  for ((i = 0; i < 10; i++)); do
+    if eval "$cmd"; then
+      return 0
+    fi
+    sleep 5
+  done
+  return 1
+}
+
+function update_apt_get() {
+  retry_apt_command "apt-get update"
+}
+
+function install_apt_get() {
+  pkgs="$@"
+  retry_apt_command "apt-get install -y $pkgs"
+}
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function install_sparkmonitor(){
+  # Install asyncio
+  retry_apt_command "${CONDA_DIRECTORY}/bin/pip install asyncio" ||
```

Comment:
Nit: just change this to `retry_command` if it is used for more than `apt` commands